lib/routes/account.js

```diff
@@ -840,6 +840,17 @@ module.exports = function (
         log.begin('Account.device', request)
         var payload = request.payload
         var sessionToken = request.auth.credentials
+        // Clients have been known to send spurious device updates,
+        // which generates lots of unnecessary database load.
+        // Don't write out the update if nothing has actually changed.
+        if (payload.id && payload.id === sessionToken.deviceId.toString('hex') &&
```

Comment:
to be clear: if it's missing the id, then we want to always process the write rather than skipping it